qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,128 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Mapping
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+
+from .base import BaseEstimatorV2
+from .containers import EstimatorPub, EstimatorPubLike, PrimitiveResult, PubResult
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options:
+    """Options for estimator."""
+
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+    def update(self, options: Options | Mapping | None = None, **kwargs):
+        """Update the options."""
+        if options is not None:
+            if isinstance(options, Mapping):
+                options_dict = options
+            elif isinstance(options, Options):
+                options_dict = options.__dict__
+            else:
+                raise TypeError(f"Type {type(options)} is not options nor Mapping class")
+            for key, val in options_dict.items():
+                setattr(self, key, val)
+
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+class Estimator(BaseEstimatorV2):
```

Comment:
There was a recommendation in the sampler statevector PR to rename it to `StatevectorSampler`. IF that goes through, this PR should do similar.

Reply:
My intent was two use case
```python
from qiskit.primitives import StatevectorEstimator
```
and
```python
from qiskit.primitives.statevector_estimator import Estimator, Options
```
the latter is assumed that users will be able to switch to other primitives simply by changing the import path.

But if `Options` is removed, this design will be dead, so I'm ok with the rename.

Reply:
You can always `import StatevectorEstimator as Estimator` if you want it to be swapable